### PR TITLE
added CPUNodeDiscovery feature gate

### PR DESCRIPTION
**Pull request by ksimon1**
This PR adds documentation for CPUNodeDiscovery feature gate
@MarSik  please review

### CODE REVIEW

File: creating-virtual-machines/virtualized-hardware-configuration.adoc [hunk 1]
```diff
@@ -161,6 +161,53 @@ spec:
 You can check list of available models
 https://github.com/libvirt/libvirt/blob/master/src/cpu_map/index.xml[here].
 
+Enabling CPUNodeDiscovery feature-gate
```
Comment (MarSik) [inline creating-virtual-machines/virtualized-hardware-configuration.adoc]: The gate name is not really that important, what about calling it Cpu compatibility enforcement or something?

File: creating-virtual-machines/virtualized-hardware-configuration.adoc [hunk 2]
```diff
@@ -161,6 +161,53 @@ spec:
...
+    cpu:
+      model: Conroe
+...
+----
+
+When CPUNodeDiscovery feature-gate is enabled, Kubevirt creates node selector with format:
+`feature.node.kubernetes.io/cpu-model-<cpuModel>`, e.g. `feature.node.kubernetes.io/cpu-model-Conroe`. 
+When VM doesn't have cpu model, then no node selector is created.
+
+To properly label the node, user can use
```
Comment (MarSik) [inline creating-virtual-machines/virtualized-hardware-configuration.adoc]: the user can also do that manually

**Comment by MarSik** [comment]
The feature gate is one thing, but I think we should create a separate page with NFD installation and explanation too.

File: creating-virtual-machines/virtualized-hardware-configuration.adoc [hunk 3]
```diff
@@ -161,12 +161,67 @@ spec:
 You can check list of available models
 https://github.com/libvirt/libvirt/blob/master/src/cpu_map/index.xml[here].
 
+Enabling cpu compatibility enforcement
+++++++++++++++++++++++++++++++++++++++
+
+To enable the CPUNodeDiscovery, user may expand the
```
Comment (MarSik) [inline creating-virtual-machines/virtualized-hardware-configuration.adoc]:
```suggestion
To enable the cpu compatibility enforcement, user may expand the
```

File: creating-virtual-machines/virtualized-hardware-configuration.adoc [hunk 4]
```diff
@@ -177,10 +232,13 @@ metadata:
 data:
   default-cpu-model: "EPYC"
 ....
+----
 
 Default CPU model is set when vmi doesn't have any cpu model. When vmi has cpu 
 model set, then vmi's cpu model is preferred. When default cpu model is not set and vmi's cpu 
 model is not set too, `host-model` will be set. Default cpu model can be changed when kubevirt is running.
+When CPUNodeDiscovery feature gate is enabled and nodes have labels with cpu models (as mentioned 
+in Enabling cpu compatibility enforcement article), Kubevirt creates node selector with default cpu model.
```
Comment (MarSik) [inline creating-virtual-machines/virtualized-hardware-configuration.adoc]: Can we make the Enabling cpu compatiblity enforcement into a hyperlink?

**Comment by yanirq** [comment]
I don't see any reference to CPU features here. 
I think we should have a general description (+ large title) before model and features section and in each section explain in more details.

File: creating-virtual-machines/virtualized-hardware-configuration.adoc [hunk 5]
```diff
@@ -161,12 +161,67 @@ spec:
 You can check list of available models
 https://github.com/libvirt/libvirt/blob/master/src/cpu_map/index.xml[here].
 
+Enabling cpu compatibility enforcement
```
Comment (yanirq) [inline creating-virtual-machines/virtualized-hardware-configuration.adoc]: CPUNodeDiscovery (as other features now) can be also added dynamically.  meaning you can `kubectl edit kubevirt-config`  the ConfigMap and the changes will apply  (double check me on the command syntax).

Comment (ksimon1) [reply creating-virtual-machines/virtualized-hardware-configuration.adoc]: but this is not article about updating config map. It is about new functionality.

**Comment by ksimon1** [comment]
@yanirq, @MarSik please review

File: creating-virtual-machines/virtualized-hardware-configuration.adoc [hunk 6]
```diff
@@ -161,12 +200,18 @@ spec:
 You can check list of available models
 https://github.com/libvirt/libvirt/blob/master/src/cpu_map/index.xml[here].
 
+When CPUNodeDiscovery feature-gate is enabled(as mentioned in link:#Enabling cpu compatibility enforcement[*Enabling cpu compatibility enforcement*])
```
Comment (yanirq) [inline creating-virtual-machines/virtualized-hardware-configuration.adoc]: the link here looks broken/malformed

File: creating-virtual-machines/virtualized-hardware-configuration.adoc [hunk 7]
```diff
@@ -134,6 +134,43 @@ spec:
...
+Labels for features user has to create by himself.
+
+To install cpu-node-labeller to cluster, run `oc apply -f cpu-node-labeller.yaml` with 
+https://github.com/ksimon1/cpu-node-labeller/blob/master/cpu-node-labeller.yaml[this deploy manifest]
+It will deploy cpu-node-labeller with cpu-model-nfd-plugin. 
+
+cpu-model-nfd-plugin uses libvirt to get all supported cpu models on host and 
+cpu-node-labeller create labels from cpu models. Then Kubevirt can schedule VM 
+on node which has support for VM cpu model.
+
```
Comment (yanirq) [inline creating-virtual-machines/virtualized-hardware-configuration.adoc]: we probably should mention if models/features are not case sensitive

Comment (ksimon1) [reply creating-virtual-machines/virtualized-hardware-configuration.adoc]: models are case sensitive afaik

File: creating-virtual-machines/virtualized-hardware-configuration.adoc [hunk 8]
```diff
@@ -134,6 +134,43 @@ spec:
...
+To properly label the node, user can use (only for cpu models)
+https://github.com/ksimon1/cpu-node-labeller[cpu-node-labeller] in combination with
+https://github.com/ksimon1/cpu-model-nfd-plugin[cpu-model-nfd-plugin] or create node labels by himself. 
+Labels for features user has to create by himself.
+
+To install cpu-node-labeller to cluster, run `oc apply -f cpu-node-labeller.yaml` with 
+https://github.com/ksimon1/cpu-node-labeller/blob/master/cpu-node-labeller.yaml[this deploy manifest]
+It will deploy cpu-node-labeller with cpu-model-nfd-plugin. 
+
+cpu-model-nfd-plugin uses libvirt to get all supported cpu models on host and
```
Comment (yanirq) [inline creating-virtual-machines/virtualized-hardware-configuration.adoc]: > cpu-model-nfd-plugin uses libvirt to get all supported cpu models 

what about features ?

Comment (ksimon1) [reply creating-virtual-machines/virtualized-hardware-configuration.adoc]: feature plugin doesn't exists yet...

File: creating-virtual-machines/virtualized-hardware-configuration.adoc [hunk 9]
```diff
@@ -253,6 +301,11 @@ Behaviour according to Policies:
 
 Full description about features and policies can be found https://libvirt.org/formatdomain.html#elementsCPU[here].
 
+When CPUNodeDiscovery feature-gate is enabled (as mentioned in link:#Enabling cpu compatibility enforcement[*Enabling cpu compatibility enforcement*]),
```
Comment (yanirq) [inline creating-virtual-machines/virtualized-hardware-configuration.adoc]: link is broken

File: creating-virtual-machines/virtualized-hardware-configuration.adoc [hunk 10]
```diff
@@ -253,6 +301,11 @@ Behaviour according to Policies:
 
 Full description about features and policies can be found https://libvirt.org/formatdomain.html#elementsCPU[here].
 
+When CPUNodeDiscovery feature-gate is enabled (as mentioned in link:#Enabling cpu compatibility enforcement[*Enabling cpu compatibility enforcement*]), 
+Kubevirt creates node selector from cpu features with format:
```
Comment (yanirq) [inline creating-virtual-machines/virtualized-hardware-configuration.adoc]: Im not sure this whole section is necessary. we talk about the behaviour above (and mention the feature gate)

File: creating-virtual-machines/virtualized-hardware-configuration.adoc (hunk 7 above)
Comment (yanirq) [reply creating-virtual-machines/virtualized-hardware-configuration.adoc]: sorry, I ment case sensitive. (btw, we should check that this is what nfd/labeller does)

Comment (ksimon1) [reply creating-virtual-machines/virtualized-hardware-configuration.adoc]: labeller doesn't do anything with names which plugins reports. It just creates labels with whatever it gets (without any modifications).

Comment (yanirq) [reply creating-virtual-machines/virtualized-hardware-configuration.adoc]: so need to check how the plugin reports

File: creating-virtual-machines/virtualized-hardware-configuration.adoc (hunk 10 above)
Comment (ksimon1) [reply creating-virtual-machines/virtualized-hardware-configuration.adoc]: But we don't talk there about label format.

File: creating-virtual-machines/virtualized-hardware-configuration.adoc (hunk 7 above)
Comment (ksimon1) [reply creating-virtual-machines/virtualized-hardware-configuration.adoc]: cpu model plugin takes models from libvirt and passes them to labeller.

File: creating-virtual-machines/virtualized-hardware-configuration.adoc [hunk 11]
```diff
@@ -134,6 +134,43 @@ spec:
...
+data:
+  feature-gates: "CPUNodeDiscovery"
+...
+----
+
+This feature-gate allows kubevirt to take VM cpu model and cpu features and create node selectors from them.
+With these node selectors, VM can be scheduled on the node which can support VM cpu model and features.
+
+To properly label the node, user can use (only for cpu models)
+https://github.com/ksimon1/cpu-node-labeller[cpu-node-labeller] in combination with
```
Comment (fabiand) [inline creating-virtual-machines/virtualized-hardware-configuration.adoc]: We should not link to personal repositories. What about proposing this and the following below to move to kubevirt org?

File: creating-virtual-machines/virtualized-hardware-configuration.adoc [hunk 12]
```diff
@@ -134,6 +134,43 @@ spec:
...
+https://github.com/ksimon1/cpu-model-nfd-plugin[cpu-model-nfd-plugin] or create node labels by himself.
```
Comment (fabiand) [inline creating-virtual-machines/virtualized-hardware-configuration.adoc]: Same herer

File: creating-virtual-machines/virtualized-hardware-configuration.adoc [hunk 13]
```diff
@@ -134,6 +134,43 @@ spec:
...
+This feature-gate allows kubevirt to take VM cpu model and cpu features and create node selectors from them.
+With these node selectors, VM can be scheduled on the node which can support VM cpu model and features.
+
+To properly label the node, user can use (only for cpu models)
+https://github.com/ksimon1/cpu-node-labeller[cpu-node-labeller] in combination with
+https://github.com/ksimon1/cpu-model-nfd-plugin[cpu-model-nfd-plugin] or create node labels by himself. 
+Labels for features user has to create by himself.
+
+To install cpu-node-labeller to cluster, run `oc apply -f cpu-node-labeller.yaml` with 
+https://github.com/ksimon1/cpu-node-labeller/blob/master/cpu-node-labeller.yaml[this deploy manifest]
```
Comment (fabiand) [inline creating-virtual-machines/virtualized-hardware-configuration.adoc]: Same here

File: creating-virtual-machines/virtualized-hardware-configuration.adoc (hunk 11 above)
Comment (ksimon1) [reply creating-virtual-machines/virtualized-hardware-configuration.adoc]: @fabiand good idea

**Comment by ksimon1** [comment]
@fabiand i am still waiting on transfering labeller + cpu plugin repo under kubevirt organization(https://groups.google.com/forum/#!topic/kubevirt-dev/jr4BVC1czzg). Do you please know who i should ping to finish this transfer?

**Comment by ksimon1** [comment]
@fabiand please can you look on this and possibly merge it?

**Pull request merged**